src/main/java/org/apache/pulsar/io/jcloud/sink/CloudStorageSinkConfig.java

```diff
@@ -77,7 +91,7 @@ public static CloudStorageSinkConfig load(Map<String, Object> map) throws IOExce
     @Override
     public void validate() {
         super.validate();
-        if (!useDefaultCredentials) {
+        if (!useDefaultCredentials && getProvider().equalsIgnoreCase(PROVIDER_AWSS3)) {
```

Comment:
should we separately check provided arguments for aws-s3 and gcs?

Reply:
this feature is only applied to `PROVIDER_AWSS3`, and when we added GCS support, it needs to bind to AWS only. And for GCS, we dont need any more check since GCS do have a fallback credential from env `GOOGLE_APPLICATION_CREDENTIALS`, refers to https://cloud.google.com/docs/authentication/getting-started#verifying_authentication